app/views/puzzles/_form.html.erb

```diff
@@ -0,0 +1,32 @@
+<%= form_for @puzzle do |f| %>
+  <p>
+    <%= f.label :name %>
+    <%= f.text_field :name %>
+  </p>
+  <p>
+    <%= f.label :description %>
+    <%= f.text_field :description %>
+  </p>
+  <p>
+    <%= f.label :level %>
+    <%= f.number_field :level, in: 0..6  %>
+  </p>
+  <p>
+    <%= f.label :producer %>
```

Comment:
Zdaje się, że ten `label` nie będzie działał (`for="producer"`, a nie masz inputu `producer`, tylko `company_id`)